src/explore-education-statistics-common/src/components/ChartsToPNG.tsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useExportButtonContext } from '@common/contexts/ExportButtonContext';
+import html2canvas, { Options as HTML2CanvasOptions } from 'html2canvas';
+import { useCallback, useState } from 'react';
+
+export type UseGenerateImage<T extends HTMLElement = HTMLDivElement> = [
+  (callback?: BlobCallback) => Promise<string | undefined>,
+  {
+    isLoading: boolean;
+  },
+];
+
+export type UseGenerateImageArgs = {
+  options?: HTML2CanvasOptions;
+  quality?: number;
+  type?: string;
+};
+
+export function useGenerateImage(
```

Comment:
I'm a bit confused about why we need `useGenerateImage` and `useCurrentPng`, they seem to do the same thing but one has type and quality args and the other has those hardcoded? Also, if they are both needed could you improve the naming so it's clearer what they're for, something like `useGenerateImageForDownload` and `useGenerateImageForClipboard`.